src/schema/server_settings.py

```diff
@@ -99,87 +99,287 @@ class Difficulty(str, Enum):
+            "type": "longtext",
+            "default": "A fantasy world",
+            "suggestOutputType": "adventure_background",
+        },
     )
 
     narrative_voice: Optional[str] = Field(
-        description="What is the narrative voice of the adventure?  Some possibilities: children’s book, young adult novel, fanfic, high literature"
+        meta_setting={
+            # Validated
+            "name": "narrative_voice",
+            "label": "Genre",
+            "description": "What is the genre of your story? E.g.: children’s book, young adult novel, fanfic, high literature.",
+            "type": "text",
+            "default": "young adult novel",
+            "suggestOutputType": "narrative_voice",
+        },
     )
 
     adventure_goal: str = Field(
         default="To rid the world of evil",
-        description="What is the ultimate goal / motivation of this story?",
+        meta_setting={
+            # Validated
+            "name": "adventure_goal",
+            "label": "Adventure Goal",
+            "description": "What is the ultimate goal / motivation of this adventure?",
+            "type": "longtext",
+            "default": "To rid the world of evil",
+            "suggestOutputType": "adventure_goal",
+        },
     )
 
     fixed_quest_arc: Optional[List[QuestDescription]] = Field(
         default=None,
-        description="If you wish for your adventure to have a fixed set of quests, define them here.",
+        meta_setting={
+            # VALIDATED
+            "name": "fixed_quest_arc",
+            "label": "Fixed Quest Arc",
+            "description": "Optional. If you wish for your adventure to have a fixed set of quests, define them here.",
+            "type": "list",
+            "listof": "object",
+            "listSchema": [
+                # TODO FUTURE this could be pulled directly from QuestDescription
+                {
+                    "name": "goal",
+                    "label": "Goal",
+                    "description": "The goal of the quest.",
+                    "type": "text",
+                },
+                {
+                    "name": "location",
+                    "label": "Location",
+                    "description": "The location of the quest.",
+                    "type": "text",
+                },
+                {
+                    "name": "description",
+                    "label": "Description",
+                    "description": "Optional description of the quest's desired characteristics.",
+                    "type": "longtext",
+                },
+            ],
+        }
     )
 
     # Quest settings
     quests_per_arc: int = Field(
         default=10,
-        description="How many quests must the player complete to finish the adventure?",
+        meta_setting={
+            # TODO: Validate int
+            "name": "quests_per_arc",
```

Comment:
Since the `name` field is repeated everywhere.. it feels like this is a good time to correct the deviation that's happened in which some of the field names on the client are different than the field names on the sever...

.. maybe we just ALWAYS use the field name in python, and we'll make sure that the. client-side does that as well. That would let us get rid fo the extra "name" field everywhere here.